snapcraft/plugins/godeps.py

```diff
@@ -0,0 +1,159 @@
+        super().__init__(name, options, project)
+        self.build_packages.append('golang-go')
+        self._gopath = os.path.join(self.partdir, 'go')
+        self._gopath_src = os.path.join(self._gopath, 'src')
+        self._gopath_bin = os.path.join(self._gopath, 'bin')
+        self._gopath_pkg = os.path.join(self._gopath, 'pkg')
+
+    def pull(self):
+        super().pull()
+
+        if self.options.go_importpath:
+            go_package = self.options.go_importpath
+        else:
+            logger.warning(
+                'Please consider setting `go-importpath` for the {!r} '
+                'part'.format(self.name))
+            go_package = os.path.basename(
+                os.path.abspath(self.options.source))
+
+        path_in_gopath = os.path.join(self._gopath_src, go_package)
+        os.makedirs(os.path.dirname(path_in_gopath), exist_ok=True)
+        if os.path.islink(path_in_gopath):
+            os.unlink(path_in_gopath)
+        os.symlink(self.sourcedir, path_in_gopath)
+
+        # Fetch and run godeps
+        logger.info('Fetching godeps...')
+        self._run(['go', 'get', 'github.com/rogpeppe/godeps'])
+
+        logger.info('Obtaining project dependencies...')
+        self._run(['godeps', '-t', '-u', os.path.join(
+            self.sourcedir, self.options.godeps_file)])
+
+    def clean_pull(self):
+        super().clean_pull()
+
+        # Remove the gopath (if present)
+        if os.path.exists(self._gopath):
+            shutil.rmtree(self._gopath)
+
+    def build(self):
+        super().build()
+
+        if self.options.go_importpath:
+            go_package = self.options.go_importpath
+        else:
+            go_package = os.path.basename(os.path.abspath(self.options.source))
+        self._run(['go', 'install', './{}/...'.format(go_package)])
+
+        install_bin_path = os.path.join(self.installdir, 'bin')
+        os.makedirs(install_bin_path, exist_ok=True)
+        os.makedirs(self._gopath_bin, exist_ok=True)
+        for binary in os.listdir(os.path.join(self._gopath_bin)):
+            # Skip godeps. It serves no purpose in production.
+            if binary == 'godeps':
+                continue
+
+            binary_path = os.path.join(self._gopath_bin, binary)
+            shutil.copy2(binary_path, install_bin_path)
+
+    def clean_build(self):
+        super().clean_build()
+
+        if os.path.isdir(self._gopath_bin):
+            shutil.rmtree(self._gopath_bin)
+
+        if os.path.isdir(self._gopath_pkg):
+            shutil.rmtree(self._gopath_pkg)
+
+    def _run(self, cmd, **kwargs):
+        env = self._build_environment()
+        return self.run(cmd, cwd=self._gopath_src, env=env, **kwargs)
+
+    def _build_environment(self):
+        env = os.environ.copy()
+        env['GOPATH'] = self._gopath
+        env['PATH'] = '{}:{}'.format(
+            os.path.join(self._gopath, 'bin'), env.get('PATH', ''))
+        linker_flags = ' '.join([
```

Comment:
can't you retrieve this from `common`, I bet you also want the stage package ones here.